src/mono/mono/mini/mini-arm64.c

```diff
@@ -5937,11 +5943,17 @@ emit_move_args (MonoCompile *cfg, guint8 *code)
 				code = emit_strfpq (code, ainfo->reg, ins->inst_basereg, GTMREG_TO_INT (ins->inst_offset));
 				break;
 			case ArgSwiftError:
-				if (ainfo->offset) {
-					code = emit_ldrx (code, ARMREG_IP0, cfg->arch.args_reg, ainfo->offset);
-					code = emit_strx (code, ARMREG_IP0, cfg->arch.swift_error_var->inst_basereg, GTMREG_TO_INT (cfg->arch.swift_error_var->inst_offset));
-				} else {
-					code = emit_strx (code, ainfo->reg, cfg->arch.swift_error_var->inst_basereg, GTMREG_TO_INT (cfg->arch.swift_error_var->inst_offset));
+				if (cfg->method->wrapper_type == MONO_WRAPPER_MANAGED_TO_NATIVE) {
+					if (ainfo->offset) {
+						code = emit_ldrx (code, ARMREG_IP0, cfg->arch.args_reg, ainfo->offset);
+						code = emit_strx (code, ARMREG_IP0, cfg->arch.swift_error_var->inst_basereg, GTMREG_TO_INT (cfg->arch.swift_error_var->inst_offset));
+					} else {
+						code = emit_strx (code, ainfo->reg, cfg->arch.swift_error_var->inst_basereg, GTMREG_TO_INT (cfg->arch.swift_error_var->inst_offset));
+					}
+				}
+				if (cfg->method->wrapper_type == MONO_WRAPPER_NATIVE_TO_MANAGED) {
+					code = emit_ldrx (code, ARMREG_IP0, cfg->arch.swift_error_var->inst_basereg, GTMREG_TO_INT (cfg->arch.swift_error_var->inst_offset));
+					code = emit_strx (code, ARMREG_R21, ARMREG_IP0, 0);
```

Comment:
Does this align with the CoreCLR implementation? It seems intuitive to load `swifterror` reg into the `SwiftError` argument, but I wonder if such cases are expected.

Reply:
For the CoreCLR implementation, RyuJIT does not load the error register's current value into the `SwiftError` arg upon method entry. The JIT implementation represents the error value by creating a `SwiftError` "pseudo-local", and converts all usages of the `SwiftError*` out parameter into usages of the local's address (see #100692). This approach allows us to be quite liberal in optimizing uses of the `SwiftError*` parameter (for example, we can promote the `SwiftError::Value` member into its own local, and enregister it), but with our implementation, I believe the initial error value upon method entry is undefined.

Reply:
@kotlarmilos do we want to remove loading `swifterror` reg into the `SwiftError` to align?

Reply:
I believe it's beneficial to align with the CoreCLR implementation. However, since it is undefined and if there are no performance implications, we can leave it as is.